### ISSUE TITLE
Automatically generate events documentation

### ISSUE THREAD

**Issue by macbre** [macbre/phantomas#729]
```
$ rgrep -E "emit\('[^']+'" modules/ extensions/ core/modules/ lib/
modules/jQuery/scope.js:            phantomas.emit('jQueryOp', type, funcName, args, contextPath, caller);
modules/jQuery/scope.js:        phantomas.emit('jQueryLoaded', version);
modules/analyzeCss/scope.js:                phantomas.emit('inlinecss', node.textContent);
modules/domQueries/scope.js:        phantomas.emit('domQuery', type, query, fnName, context, hasNoResults); // @desc DOM query has been made
modules/domComplexity/scope.js:            phantomas.emit('DOMids', ids);
modules/domComplexity/scope.js:                                    phantomas.emit('imagesScaledDown', {
extensions/screenshot/screenshot.js:			phantomas.emit('screenshot', path);
extensions/pageSource/pageSource.js:			phantomas.emit('pageSource', path);
extensions/filmStrip/filmStrip.js:			phantomas.emit('filmStrip', path, ts);
extensions/scroll/scroll.js:			phantomas.emit('scroll', scrollOffset);
core/modules/navigationTiming/scope.js:        phantomas.emit('milestone', eventName);
core/modules/navigationTiming/scope.js:        emit('domReady');
core/modules/navigationTiming/scope.js:                emit('domInteractive');
core/modules/navigationTiming/scope.js:                emit('domComplete');
core/modules/timeToFirstByte/timeToFirstByte.js:			phantomas.emit('milestone', 'responseEnd');
core/modules/requestsMonitor/requestsMonitor.js:		phantomas.emit('send', request); // @desc request has been sent
core/modules/requestsMonitor/requestsMonitor.js:			phantomas.emit('base64recv', entry, resp); // @desc base64-encoded "response" has been received
core/modules/requestsMonitor/requestsMonitor.js:			phantomas.emit('recv' , entry, resp); // @desc response has been received
lib/index.js:			events.emit('init', browser, page);
lib/index.js:			events.emit('report');
lib/browser.js:        this.events.emit('consoleLog', msg);
lib/browser.js:        this.events.emit('jserror', lines[0].trim(), lines.slice(1));
lib/browser.js:        this.events.emit('request', request);
lib/browser.js:        this.events.emit('response', response);
lib/browser.js:        this.events.emit('loaded', this.page);
lib/browser.js:        this.events.emit('metrics', metrics);
lib/browser.js:    this.events.emit('close');
```

```
$ rgrep --no-filename -oE "emit\('([^'])+'" modules/ extensions/ core/modules/ lib/ | sed 's/emit(//g' | sed "s/'//g" | sort | uniq
base64recv
close
consoleLog
domComplete
DOMids
domInteractive
domQuery
domReady
filmStrip
imagesScaledDown
init
inlinecss
jQueryLoaded
jQueryOp
jserror
loaded
metrics
milestone
pageSource
recv
report
request
response
screenshot
scroll
send
```

**Comment by macbre** [comment]
Let's focus on "core" events for now:

```
$ rgrep -E "\.emit\('[^']+'" core/modules/ lib/  | sort
core/modules/navigationTiming/scope.js:        phantomas.emit('milestone', eventName); // @desc Page loading milestone has been reached: domInteractive, domReady and domComplete
core/modules/requestsMonitor/requestsMonitor.js:			phantomas.emit('base64recv', entry, resp); // @desc base64-encoded "response" has been received
core/modules/requestsMonitor/requestsMonitor.js:			phantomas.emit('recv' , entry, resp); // @desc response has been received
core/modules/requestsMonitor/requestsMonitor.js:		phantomas.emit('send', request); // @desc request has been sent
core/modules/timeToFirstByte/timeToFirstByte.js:			phantomas.emit('responseEnd', entry, res); // @desc The first response (that was not a redirect) fully received
lib/browser.js:        this.events.emit('consoleLog', msg); // @desc `console.log` has been called in page's scope
lib/browser.js:        this.events.emit('jserror', lines[0].trim(), lines.slice(1)); // @desc Emitted when an uncaught exception happens within the page
lib/browser.js:        this.events.emit('loaded', this.page);  // @desc Emitted when the page has been fully loaded
lib/browser.js:        this.events.emit('metrics', metrics); // @desc Emitted when Chromuim's page.metrics() has been called
lib/browser.js:        this.events.emit('request', request); // @desc Emitted when page is about to send HTTP request
lib/browser.js:        this.events.emit('response', response); // @desc Emitted when page received a HTTP response
lib/browser.js:    this.events.emit('close'); // @desc Chromium has been closed
lib/index.js:			await events.emit('beforeClose', page); // @desc Called before the Chromium (and all of its pages) is closed
lib/index.js:			await events.emit('init', page, browser); // @desc Browser's scope and modules are set up, the page is about to be loaded
lib/index.js:			await events.emit('report'); // @desc Called just before the phantomas results are returned to the caller
```

**Comment by macbre** [comment]
Resolved by #730
